nemo/collections/nlp/modules/common/megatron/module.py

```diff
@@ -43,12 +43,13 @@ class MegatronModule(torch.nn.Module):
     """Megatron specific extensions of torch Module with support
     for pipelining."""
 
-    def __init__(self, share_token_embeddings=True):
+    def __init__(self, config: ModelParallelConfig = None, share_token_embeddings=True):
         if not HAVE_MEGATRON_CORE:
             raise ImportError(
                 "megatron-core was not found. Please see the NeMo README for installation instructions: https://github.com/NVIDIA/NeMo#megatron-gpt."
             )
         super(MegatronModule, self).__init__()
+        self.config = config
```

Comment:
We use TransformerConfig for config in megatron-core, and MPConfig here
I'm afraid this might cause confusions...

Reply:
Agreed. I tried to update to TransformerConfig for NeMo but there's too much scope so I think we need to stick with this middleground and just get everything on mcore asap.